poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -564,16 +559,127 @@ public void saveSensorAssetsToFlowgate(Map<String, Asset> exsitingSensorAssets,
                   asset.setMountingSide(sensorMountingSide.get(sensor.getPosition().toUpperCase()));
                }
                //save
-               assetsNeedToSave.add(asset);
+               newAssetsNeedToSave.add(asset);
             }
          }
-         restClient.saveAssets(assetsNeedToSave);
+         restClient.saveAssets(oldAssetsNeedToupdate);
+         if(!newAssetsNeedToSave.isEmpty()) {
+            List<Asset> sensorAlreadySaved = Arrays.asList(restClient.saveAssets(newAssetsNeedToSave).getBody());
+            List<Asset> pduAssetNeedToUpdate = updatePduMetricformular(sensorAlreadySaved,pduAssetMap);
+            restClient.saveAssets(pduAssetNeedToUpdate);
+         }
          offset += limit;
       }
+   }
+
+   public List<Asset> updatePduMetricformular(List<Asset> sensorAssets, Map<String,Asset> pduIdAndAssetMap){
+      List<Asset> pduAssets = new ArrayList<Asset>();
+      for(Asset sensorAsset : sensorAssets) {
+         com.vmware.flowgate.common.model.Parent parent = sensorAsset.getParent();
+         if(parent == null) {
+            continue;
+         }
+         String pduId = parent.getParentId();
+         Asset pduAsset = pduIdAndAssetMap.get(pduId);
+         int position = sensorAsset.getCabinetUnitPosition();
+         String positionInfo = FlowgateConstant.DEFAULT_CABINET_UNIT_POSITION;
+         Map<String,String> sensorAssetJustfication = sensorAsset.getJustificationfields();
+         if(position != 0) {
+            positionInfo = String.valueOf(position);
```

Comment:
how about use the string like rackUnit+position ?

Reply:
ok